Feature/detail views

femans opened the pull request:
work done, big work done!
from the notes: 
* Documents-  pdf viewer added, the rest says: "can't display"
* Download button - done
* Metadata? - added - except image page, because additional styling is needed
* Audio player: long story:

audio player has been refactored largely and is now available on every page.
- it has a 'seeker' (and if you don't know what it is, try out the player.)
- it stops if the videoplayer begins to play
- audio player has spinner while it is loading

TODO still:
- some styling is still needed. The v-bottom-sheet was not usable for the player (even though vuetify advertises with it) because it blocks the entire page. So I replaced it with a v-sheet, which may use some minor styling work.
- more testing. It has been tested well, but it needs more mileage
- imageDetail page needs metadata and header. I couldn't because of styling, i could not get it right.
- Directory page, I couldn't do it because I have no data to display.

dokterbob commented on package.json:
```diff
@@ -9,6 +9,7 @@
     "lint": "vue-cli-service lint"
   },
   "dependencies": {
+    "axios": "^0.21.4",
```
Which component requires axios? We're currently using the browser native `fetch()` with polyfill and it would be great if we could stick with that (but it's not essential).

femans replied:
download button; i did not have a lot of time for that, i plucked a snippet from SO to do that, but I agree it is clunky and should be improved. I will take a look.

femans replied:
fixed it in next commit

dokterbob commented on src/components/AudioPlayer.vue:
```diff
@@ -0,0 +1,186 @@
+      this.$data.loading = true;
+
+      this.sound = new Howl({
+        src: [`https://gateway.ipfs.io/ipfs/${fileObject.hash}`],
```
TODO: Add an issue to make gateway configurable.

We want this to be a dynamic variable, possibly from VueX, which takes the local IPFS node when available and possibly our own gateway over the public one.

femans replied:
yeah but that is outside of scope for this MR

dokterbob commented on src/helpers/fileHelper.js:
```diff
@@ -0,0 +1,8 @@
+export function getFileExtension(fileName) {
```
Many files do not have a filename and, hence, no extension! Other files have multiple references ('names') and so this is not a reliable way to do this.

However, the API supplies a `mimetype` field which, together with something like [mime-file-extension](https://www.npmjs.com/package/mime-file-extension) can be used to derive extensions.

The current implementation is good enough for testing purposes but we should most definitely create an issue referring this before this one is merged so we won't forget about it.

This affects a considerable amount of items.

femans replied:
The reason for this implementation is that howler.js itself, the chosen player, uses the extension to derive the type. The mimetype does not give accurate/precise enough information, to my knowledge. E.g. audio/mpeg can mean a host of different formats. 
The filetype is only necessary for the audio player. We could hide the player for files that do not have sufficient information available. I would suggest to push that to another MR.

dokterbob commented on src/router/index.js:
```diff
@@ -15,20 +15,22 @@ const routes = [
   },
   {
-    path: '/search/detail',
+    path: '/search/detail/:fileType',
```
We can merge this now but we should add an issue splitting out the detail views for various types to their respective URL's. Some of these viewers require a sizeable chunk of JS (e.g. Video.js) and having lazy-loading on them is more than a luxury.

dokterbob commented on src/views/Detail.vue:
```diff
@@ -64,199 +64,35 @@
-                </div>
-              </div>
-            </v-row>
-          </v-sheet>
```
Yay for getting rid of all the Lorum Ipsum!

dokterbob reviewed (changes requested):
Great! Super! Great progess!

Apart from the music player, which I can't get to work, it's ready to merge - just please let's be mindful of known issues - ideally they would be references with exact code references so future fixes don't require additional bug hunting.

1. Issues need to be made for various review comments, before merging.
2. Issues need to be added also for TODO items. It would be great if you found time also to discuss (design) requirements with @vandrongelen in a call or so.
2. Audio player doesn't seem to play for known available content (tried both list and detail views). Reference content (first hit): https://ipfs-search.com/#/search?kind=audio&search=stones%20satisfaction Tested with Brave - no errors in browser.

Thanks! :D

dokterbob commented:
I double checked after doing `npm install` and I still can't get the audio player to work.

Detail URL: http://localhost:8080/search/detail/audio?q=satisfaction%20stones&last_seen&size&type=audio&page=0#QmZezVi9Pwnaa6Ym94FYn8qXKbT3fNDG2gv529m4wbc5jw
List URL: http://localhost:8080/search?q=satisfaction%20stones&last_seen&size&type=audio&page=
[package-lock.json.zip](https://github.com/ipfs-search/dweb-search-frontend/files/7190608/package-lock.json.zip)
0
Package lock attached for reference.

femans commented:
Weird! Moeten we verder onderzoeken. Kun je beschrijven wat er gebeurt of
niet gebeurt?

Op za 18 sep. 2021 22:38 schreef Mathijs de Bruin ***@***.***
>:

> I double checked after doing npm install and I still can't get the audio
> player to work.
>
> Detail URL:
> http://localhost:8080/search/detail/audio?q=satisfaction%20stones&last_seen&size&type=audio&page=0#QmZezVi9Pwnaa6Ym94FYn8qXKbT3fNDG2gv529m4wbc5jw
> List URL:
> http://localhost:8080/search?q=satisfaction%20stones&last_seen&size&type=audio&page=
> package-lock.json.zip
> <https://github.com/ipfs-search/dweb-search-frontend/files/7190608/package-lock.json.zip>
> 0
> Package lock attached for reference.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ipfs-search/dweb-search-frontend/pull/24#issuecomment-922370395>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEYZOYXJBC3ENPRIJX4ML3UCT2FDANCNFSM5EHR26OQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

dokterbob commented:
Ik druk vanuit de detail op play en dan… gebeurd er nix.

Vanuit de lijst weergave gebeurd er dit:



Ook geen error in de console.

Ik heb in Safari en Brave getest.

> Op 19 sep. 2021, om 09:00 heeft Frido Emans ***@***.***> het volgende geschreven:
> 
> 
> Weird! Moeten we verder onderzoeken. Kun je beschrijven wat er gebeurt of
> niet gebeurt?
> 
> Op za 18 sep. 2021 22:38 schreef Mathijs de Bruin ***@***.***
> >:
> 
> > I double checked after doing npm install and I still can't get the audio
> > player to work.
> >
> > Detail URL:
> > http://localhost:8080/search/detail/audio?q=satisfaction%20stones&last_seen&size&type=audio&page=0#QmZezVi9Pwnaa6Ym94FYn8qXKbT3fNDG2gv529m4wbc5jw
> > List URL:
> > http://localhost:8080/search?q=satisfaction%20stones&last_seen&size&type=audio&page=
> > package-lock.json.zip
> > <https://github.com/ipfs-search/dweb-search-frontend/files/7190608/package-lock.json.zip>
> > 0
> > Package lock attached for reference.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/ipfs-search/dweb-search-frontend/pull/24#issuecomment-922370395>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/AAEYZOYXJBC3ENPRIJX4ML3UCT2FDANCNFSM5EHR26OQ>
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> > or Android
> > <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
> >
> >
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/ipfs-search/dweb-search-frontend/pull/24#issuecomment-922432524>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAFRVUIKWNNE7D3YL3RL7LUCWKCTANCNFSM5EHR26OQ>.
> Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>

femans commented:
Ik vermoed dat het komt doordat de file geen title heeft. Ik zal die
mimetype conversion proberen, zoals je had voorgesteld, ik had al met een
hardcoded mp3 geprobeerd en dan doen sommige dingen het wel die het anders
niet doen.

Nog steeds blijven loading times heel vreemd, downloads gaan soms
razendsnel terwijl de player eindeloos blijft laden op dezelfde file.

Zoals gezegd nog uitgebreid testen kan geen kwaad.

Op zo 19 sep. 2021 14:05 schreef Mathijs de Bruin ***@***.***
>:

> Ik druk vanuit de detail op play en dan… gebeurd er nix.
>
> Vanuit de lijst weergave gebeurd er dit:
>
>
>
> Ook geen error in de console.
>
> Ik heb in Safari en Brave getest.
>
> > Op 19 sep. 2021, om 09:00 heeft Frido Emans ***@***.***> het volgende
> geschreven:
> >
> >
> > Weird! Moeten we verder onderzoeken. Kun je beschrijven wat er gebeurt of
> > niet gebeurt?
> >
> > Op za 18 sep. 2021 22:38 schreef Mathijs de Bruin ***@***.***
> > >:
> >
> > > I double checked after doing npm install and I still can't get the
> audio
> > > player to work.
> > >
> > > Detail URL:
> > >
> http://localhost:8080/search/detail/audio?q=satisfaction%20stones&last_seen&size&type=audio&page=0#QmZezVi9Pwnaa6Ym94FYn8qXKbT3fNDG2gv529m4wbc5jw
> > > List URL:
> > >
> http://localhost:8080/search?q=satisfaction%20stones&last_seen&size&type=audio&page=
> > > package-lock.json.zip
> > > <
> https://github.com/ipfs-search/dweb-search-frontend/files/7190608/package-lock.json.zip
> >
> > > 0
> > > Package lock attached for reference.
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <
> https://github.com/ipfs-search/dweb-search-frontend/pull/24#issuecomment-922370395
> >,
> > > or unsubscribe
> > > <
> https://github.com/notifications/unsubscribe-auth/AAEYZOYXJBC3ENPRIJX4ML3UCT2FDANCNFSM5EHR26OQ
> >
> > > .
> > > Triage notifications on the go with GitHub Mobile for iOS
> > > <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> >
> > > or Android
> > > <
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> >.
> > >
> > >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub <
> https://github.com/ipfs-search/dweb-search-frontend/pull/24#issuecomment-922432524>,
> or unsubscribe <
> https://github.com/notifications/unsubscribe-auth/AAAFRVUIKWNNE7D3YL3RL7LUCWKCTANCNFSM5EHR26OQ
> >.
> > Triage notifications on the go with GitHub Mobile for iOS <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android <
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ipfs-search/dweb-search-frontend/pull/24#issuecomment-922462952>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEYZO374IIRGFSDD3WIIYLUCXGXDANCNFSM5EHR26OQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

femans commented:
I was wrong, it was simply a bug. I fixed it, and added mimetype
conversion, but that is not ideal either. I am going to try to replace it
with file header detection.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

On Sun, Sep 19, 2021 at 2:56 PM Frido Emans ***@***.***> wrote:

> Ik vermoed dat het komt doordat de file geen title heeft. Ik zal die
> mimetype conversion proberen, zoals je had voorgesteld, ik had al met een
> hardcoded mp3 geprobeerd en dan doen sommige dingen het wel die het anders
> niet doen.
>
> Nog steeds blijven loading times heel vreemd, downloads gaan soms
> razendsnel terwijl de player eindeloos blijft laden op dezelfde file.
>
> Zoals gezegd nog uitgebreid testen kan geen kwaad.
>
> Op zo 19 sep. 2021 14:05 schreef Mathijs de Bruin <
> ***@***.***>:
>
>> Ik druk vanuit de detail op play en dan… gebeurd er nix.
>>
>> Vanuit de lijst weergave gebeurd er dit:
>>
>>
>>
>> Ook geen error in de console.
>>
>> Ik heb in Safari en Brave getest.
>>
>> > Op 19 sep. 2021, om 09:00 heeft Frido Emans ***@***.***> het volgende
>> geschreven:
>> >
>> >
>> > Weird! Moeten we verder onderzoeken. Kun je beschrijven wat er gebeurt
>> of
>> > niet gebeurt?
>> >
>> > Op za 18 sep. 2021 22:38 schreef Mathijs de Bruin ***@***.***
>> > >:
>> >
>> > > I double checked after doing npm install and I still can't get the
>> audio
>> > > player to work.
>> > >
>> > > Detail URL:
>> > >
>> http://localhost:8080/search/detail/audio?q=satisfaction%20stones&last_seen&size&type=audio&page=0#QmZezVi9Pwnaa6Ym94FYn8qXKbT3fNDG2gv529m4wbc5jw
>> > > List URL:
>> > >
>> http://localhost:8080/search?q=satisfaction%20stones&last_seen&size&type=audio&page=
>> > > package-lock.json.zip
>> > > <
>> https://github.com/ipfs-search/dweb-search-frontend/files/7190608/package-lock.json.zip
>> >
>> > > 0
>> > > Package lock attached for reference.
>> > >
>> > > —
>> > > You are receiving this because you authored the thread.
>> > > Reply to this email directly, view it on GitHub
>> > > <
>> https://github.com/ipfs-search/dweb-search-frontend/pull/24#issuecomment-922370395
>> >,
>> > > or unsubscribe
>> > > <
>> https://github.com/notifications/unsubscribe-auth/AAEYZOYXJBC3ENPRIJX4ML3UCT2FDANCNFSM5EHR26OQ
>> >
>> > > .
>> > > Triage notifications on the go with GitHub Mobile for iOS
>> > > <
>> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
>> >
>> > > or Android
>> > > <
>> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
>> >.
>> > >
>> > >
>> > —
>> > You are receiving this because you commented.
>> > Reply to this email directly, view it on GitHub <
>> https://github.com/ipfs-search/dweb-search-frontend/pull/24#issuecomment-922432524>,
>> or unsubscribe <
>> https://github.com/notifications/unsubscribe-auth/AAAFRVUIKWNNE7D3YL3RL7LUCWKCTANCNFSM5EHR26OQ
>> >.
>> > Triage notifications on the go with GitHub Mobile for iOS <
>> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android <
>> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>>
>> >
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ipfs-search/dweb-search-frontend/pull/24#issuecomment-922462952>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAEYZO374IIRGFSDD3WIIYLUCXGXDANCNFSM5EHR26OQ>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>>
>>
>

dokterbob reviewed (approved):
Ready for merge! 

Please rebase to clean up commit history and please create issues for various TODO's.

Thanks! :D

femans commented:
> Ready for merge!
> 
> Please rebase to clean up commit history and please create issues for various TODO's.
> 
> Thanks! :D

Are you sure you don't mean a squash merge? (because rebase really doesn't clean up commit history)
